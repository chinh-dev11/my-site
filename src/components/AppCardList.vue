<template>
  <ul :class="$style.cards">
    <li
      v-for="item in items"
      :key="item.header"
    >
      <BaseCard :item="item" />
    </li>
  </ul>
</template>

<script>
    import BaseCard from './BaseCard';

    export default {
        components: {
            BaseCard
        },
        props: {
            items: {
                type: Array,
                default () {
                    return [];
                }
            }
        }        
    }
</script>

<style lang="scss" module>
.cards {
    li {
        border-radius: 0.5em;
        background-color: #fff;
        padding: 1em;
        margin-bottom: 1em;
    }
}
</style>