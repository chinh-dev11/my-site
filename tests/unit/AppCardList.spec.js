import { shallowMount } from '@vue/test-utils'
import AppCardList from '@/components/AppCardList.vue'

describe('AppCardList', () => {
  it('renders li for each item in props.items', () => {
    const items = [{}, {}]
    const wrapper = shallowMount(AppCardList, {
      propsData: { items }
    })
    expect(wrapper.findAll('li')).toHaveLength(items.length)
  })

  it('matches snapshot', () => {
    const items = [
      {
        order: 3
      },
      {
        order: 2
      }
    ]
    const wrapper = shallowMount(AppCardList, {
      propsData: { items }
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.html()).toMatchInlineSnapshot(`
      <ul>
        <li>
          <basecard-stub item="[object Object]"></basecard-stub>
        </li>
        <li>
          <basecard-stub item="[object Object]"></basecard-stub>
        </li>
      </ul>
    `)
  })
})
