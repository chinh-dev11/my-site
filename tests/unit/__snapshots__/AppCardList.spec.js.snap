// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppCardList matches snapshot 1`] = `
<ul>
  <li>
    <basecard-stub item="[object Object]"></basecard-stub>
  </li>
  <li>
    <basecard-stub item="[object Object]"></basecard-stub>
  </li>
</ul>
`;
